### PR TITLE
Add warning decorator

### PR DESCRIPTION
**Pull request by Ibrahim2750mi**

### CODE REVIEW

File: arcade/utils.py [hunk 1]
```diff
@@ -4,12 +4,29 @@
...
 """
+import functools
 import platform
 import sys
+import warnings
 from typing import Tuple
 from pathlib import Path
 
 
+class PerformanceWarning(Warning):
```
Comment (einarf) [inline arcade/utils.py:15]: `Type[Warning]`

Comment (Ibrahim2750mi) [reply arcade/utils.py:15]: Didn't saw this wait

**Comment by Ibrahim2750mi** [comment]
It is still failing? @einarf

File: arcade/utils.py [hunk 2]
```diff
@@ -4,12 +4,29 @@
...
+from typing import Tuple, Type
 from pathlib import Path
 
 
+class PerformanceWarning(Type[Warning]):
+    """Use this for issuing performance warnings."""
+    pass
+
+
+def warning(message: str, warning_type: Warning):
```
Comment (einarf) [inline arcade/utils.py]: `Type[Warning]` here, sorry!

Comment (Ibrahim2750mi) [reply arcade/utils.py]: updated in https://github.com/pythonarcade/arcade/pull/1615/commits/f7f1e15006db232aea5b33ca1c8c50edd2a8fb85

**Pull request merged**